src/item.h

```diff
@@ -876,6 +910,14 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
          * Callback when contents of the item are affected in any way other than just processing.
          */
         void on_contents_changed();
+
+         /**
+          * Callback immediately **before** an item is damaged
+          * @param dmg type of damage (or DT_NULL)
+          * @param qty maximum damage that will be applied (constrained by @ref max_damage)
+          */
+        void on_damage( damage_type dmg, int qty );
```

Comment:
Why is order of `qty` and `dmg` (`dt` would be better) here different from the other functions?